docs/openapi3.yaml

```diff
@@ -1096,6 +1096,70 @@ paths:
           $ref: '#/components/responses/ResponseScriptExecutionForbidden'
         '404':
           description: Expected response to a non-existing script execution
+  /openapi/transform:
+    post:
+      summary: Transform an OpenAPI File into connections and components
+      tags:
+        - openapi
+      operationId: openapiTransform
+      security:
+        - bearerAuth:
+            - OPENAPI_READ
+      requestBody:
+        description: The openapi file to be transformed
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+          multipart/formData:
+            schema:
+              type: string
+              format: binary
+      responses:
+        '200':
+          description: Expected response to a valid request
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  components:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/ComponentDto'
+                  connections:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/ConnectionDto'
+        '401':
+          $ref: '#/components/responses/TokenResponse'
+        '403':
+          $ref: '#/components/responses/ResponseOpenApiForbidden'
+        '404':
+          description: Expected response to an unexisting openapi file from a server
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+              example:
+                timestamp: '2021-02-09T07:57:01.062+0000'
+                status: 404
+                error: Not Found
+                message: Unable the find the openapi specification
+                path: /api/openapi/
+        '500':
+          description: Expected response to an internal error.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+              example:
+                timestamp: '2021-02-09T07:57:01.062+0000'
+                status: 500
+                error: ComponentParser Exception
+                message: The openapi title is missing
+                path: /api/openapi/
```

Comment:
should not occur, this should be captured as a 400 with a decent error message.
Is this already handled in the code? Otherwise have a look at ControllerAdvice on how to handle Exceptions in Spring